app/routes/route_utilities.py

```diff
@@ -0,0 +1,63 @@
+from flask import abort, make_response
+from app.db import db
+from app.models.goal import Goal
+from app.models.task import Task
+import requests
+import os
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"message": f"{cls.__name__} {model_id} is invalid, should be an int data type"}, 400))
+
+    query = db.select(cls).where(cls.id == model_id)
+    model = db.session.scalar(query)
+
+    if not model:
+        abort(make_response({"message": f"{cls.__name__} {model_id} is not found"}, 404))
+    
+    return model
+
+def create_model(cls, model_data):
+    try:
+        new_model = cls.from_dict(model_data)
+        
+    except KeyError as error:
+        abort(make_response({'details': 'Invalid data'}, 400))
```

Comment:
It would be nice if we could provide more details about the invalid request body so that the client sending the request knows what about the body to fix.

```py
response = {"details": f"Invalid data: missing {error.args[0]}"}
```